app/assets/templates/posts/userpost.html

```diff
@@ -1,10 +1,27 @@
 <div class="container">
+  
   <div class="row">
     <h1>Show All Posts by One User</h1>
-
     <h4>Title: {{ post.title }}</h4>
     <h4>Content: {{ post.content }}</h4>
-    <h4>Category: {{ post.category ? post.category : 'null' }}</h4>
```

Comment:
Idea: If you intended to only show the category if it has one, perhaps you could use [ng-show](https://docs.angularjs.org/api/ng/directive/ngShow)?